### ISSUE TITLE
[Bounty] Atari Jaguar CD Support

### ISSUE THREAD

**Issue by klepp0906** [libretro/virtualjaguar-libretro#41]
~~If its even reasonable im considering opening a bounty as theres currently no way to emulate the cd apart from project tempest which was last updated in 2004~~

~~unsure how much work is involved, or if its already been talked about or potentially planned etc~~

Hoping to get virtualjaguar supporting the small Jaguar CD library.   There is very little left missing from retroarch, and this is one of the items.   Nothing out there currently emulates it apart from Project Tempest and its pretty broken on about half the titles aaaand we have no project tempest core :P 

A separate core is just the same, but i imagine more work, pick your poison :)   

Ideally would at least support cdi, as all my dumps are converted to such due to that being what project tempest called for (and i dont think they can be converted to chd or some such).  Another stipulation that can go either way.  Just a strong preference as I imagine thats the format most people have due to the aforementioned.

**Comment by Cee123** [comment]
I'll bet we could get a bounty going on this as well. Even though my previous bounty to fix the issues with this core and the unplayability of most games didn't get any results. :(

**Comment by klepp0906** [comment]
If you feel like opening a bounty, ill contribute and spread the word a bit.   I'm a bit busy atm (daughters bday party) but will eventually get around to it if someone else doesnt first.

This and CD-i are all thats left on the board for RA i think, apart from the modern stuff.  (cd-i is techncially doable by mame but saying its clunky is being generous)

**Comment by Cee123** [comment]
Yeah we could start a bounty. Not sure how it will go, can't even get someone to fix those Jaguar games that don't even work.

You are right. Jaguar and CD-i are the only things really missing from Retroarch nowadays. But 3DO, Saturn and Nintendo DS could also be a lot better too imo. I tried to get CD-i to work on Retroarch but I haven't had any luck using either Mame or MESS - it just crashes.

**Comment by klepp0906** [comment]
Bounty is up.

**Comment by Spikestuff** [comment]
Dunno if this'll help you out or not but BizHawk which also uses VirtualJaguar does have [JagCD support](https://tasvideos.org/5658M). - https://github.com/TASEmulators/BizHawk/tree/master/waterbox/virtualjaguar/src

**Comment by YoshiRulz** [comment]
[BountySource went bust](https://news.ycombinator.com/item?id=38419134), so there is no bounty.

**Comment by JoeMatt** [comment]
**Status: Jaguar CD support is implemented and merged.** This issue predates the work by six years, so a summary of where it landed.

CD emulation went in via #201 and a long tail of follow-ups, and is on `develop` now (shipping in nightlies, and in releases from v2.3.x onward).

**Formats:** CUE/BIN, CHD, and CDI — CDI was the explicit ask here, and `test/tools/cue2cdi` also exists to convert a CUE/BIN set into the CDI layout.

**Two boot paths:**
- **HLE** — the CD BIOS calls are emulated directly. No BIOS dump needed.
- **Real BIOS** — runs the actual Jaguar CD BIOS with the authentication step bypassed.

**Current state**, from `docs/cd-boot-matrix.md` (regenerated per build, not hand-maintained): **22/22 rows reach `GAME_CODE`** — 10 commercial CUE/BIN titles plus a CDI image, each in both HLE and real-BIOS mode. Baldies, Battle Morph, BrainDead 13, Dragon's Lair, Highlander, Hover Strike, Iron Soldier 2, Primal Rage, Space Ace, Myst.

That covers a large majority of the commercial Jaguar CD library. Titles boot, run game code, stream data and play CD audio; a fair number have been played through by hand rather than only measured headlessly.

**Not perfect** — remaining known issues are tracked individually in `docs/cd-known-issues.md` and as their own tickets, e.g. #230 (CDI images not produced by our own converter can compute a wrong session-2 offset). Those are specific bugs now, not "no CD support".

On the bounty: [BountySource shut down](https://news.ycombinator.com/item?id=38419134), as @YoshiRulz noted, so there is nothing to collect and nothing owed. The work happened anyway.

@klepp0906 @Cee123 — if you still have your CDI dumps, a nightly is worth a try. If one fails to load, that is #230's shape and a fresh issue with the image's size and the `[CD-BOOTSTUB]` log lines would be genuinely useful.

Suggest closing this as implemented, with specific breakage tracked in its own tickets.
